html.js

```diff
@@ -18,6 +18,8 @@ const Html = ({ body }) => {
 
         <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
         <link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,500,600,900" rel="stylesheet" />
+  
+        <link rel="icon" type="img/png" href={ prefixLink('/assets/images/favicon.png') } />
```

Comment:
Your `html.js` is rendered from `pages/index.js` so your path has to be relative to `pages/index.js`.  However, using `../assets/images/favicon.png` here doesn't work still for some reason, so I would move the favicon out of `assets/images` to the `/pages` folder instead and change the path to be `/favicon.png`